### DISCUSSION TITLE
Is aptos-std::event deprecated now?

### DISCUSSION THREAD

**Discussion by 0x-j** [aptos-labs/aptos-developer-discussions#313] [Questions] [answered]
### Discord user ID

alice_or_bob

### Describe your question in detail.

From discord

> Is aptos-std::event deprecated now? My previous contract used aptos-std::event and I need to redeploy a contract recently, can I continue to redeploy the contract using the previous code? Or is it more recommended to replace aptos-std::event with aptos-framework::event?

### Which operating system are you using?

macOS

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by 0x-j** [comment, marked as answer]
Yes, it's more recommended to use event from framework as it simplify a lot of things, most importantly no need for event store.

See this example on [defining event](https://github.com/aptos-labs/move-by-examples/blob/main/fungible-asset-launchpad/aptos/move/sources/launchpad.move#L11-L21) and [emitting event](https://github.com/aptos-labs/move-by-examples/blob/main/fungible-asset-launchpad/aptos/move/sources/launchpad.move#L83-L92). 

You can learn more about it from [doc](https://aptos.dev/en/network/blockchain/events#module-events), we call the new event `module event`

**Comment by 99Kies** [comment]
@0xaptosj We have a requirement to update our contract, mainly because we want to replace the owner/admin address of the contract. Is it be easier to rotate the key so that we don't have to update the contract? But I don't know if there are any other details that we need to pay attention to, and I don't know if it will affect the related resource query function.

**Reply by 0x-j** [reply]
yeah i think rotating key would make sense in this case, it has basically no impact on anything depends on the contract, everything stays the same, it's just a different private key.
